packages/stateful/voting-module-adapter/adapters/CwdVotingCw721Staked/components/ProfileCardMemberInfo/StakingModal.tsx

```diff
@@ -0,0 +1,481 @@
+        if (Math.abs(walletStakedBalance - amountToUnstake) <= 1) {
+          amountToUnstake = walletStakedBalance
+        }
+
+        try {
+          await doUnstake({
+            tokenIds,
+          })
+
+          // New balances will not appear until the next block.
+          await awaitNextBlock()
+
+          refreshBalances()
+          refreshTotals()
+          refreshClaims?.()
+          refreshDaoVotingPower()
+
+          setTokenIds([])
+          toast.success(
+            `Unstaked ${tokenIds.length} $${governanceTokenInfo.symbol}`
+          )
+
+          // Close once done.
+          onClose()
+        } catch (err) {
+          console.error(err)
+          toast.error(processError(err))
+        } finally {
+          setStakingLoading(false)
+        }
+
+        break
+      }
+      default:
+        toast.error('Internal error while staking. Unrecognized mode.')
+    }
+  }
+
+  const onNftClick = (nft: NftCardInfo) => {
+    let tokenId = getIdForNft(nft)
+    let _tokenIds = [...tokenIds]
+
+    if (_tokenIds.includes(tokenId)) {
+      _tokenIds = _tokenIds.filter((_tokenId) => {
+        return _tokenId !== tokenId
+      })
+    } else {
+      _tokenIds.push(getIdForNft(nft))
+    }
+
+    setTokenIds(_tokenIds)
+  }
+
+  const onDeselectAll = () => {
+    setTokenIds([])
+  }
+
+  const onSelectAll = () => {
+    let _tokenIds =
+      !nfts.loading && !nfts.errored
+        ? nfts.data.map((nft) => {
+            return getIdForNft(nft)
+          })
+        : []
+
+    setTokenIds(_tokenIds)
+  }
+
+  const getIdForNft = (nft: NftCardInfo) => {
+    return nft.tokenId
+  }
+
+  const updateStakingMode = useCallback(
+    (newMode: StakingMode) => {
+      setTokenIds([])
+
+      let _header = Object.assign({}, header)
+
+      if (walletStakedValue !== undefined && walletStakedValue > 0) {
+        _header.title = t('title.manageStaking')
+      } else {
+        _header.title = t('title.stakingModeNfts.stake')
+      }
+
+      switch (newMode) {
+        case StakingMode.Stake:
+          setActionLabel(t('button.stake'))
+          _header.subtitle = t('title.stakingModeNfts.stakeHeaderSubtitle')
+          setHeader(_header)
+          setNfts(
+            loadingWalletUnstakedNfts ??
+              ({ loading: false, errored: true } as LoadingDataWithError<
+                NftCardInfo[]
+              >)
+          )
+          break
+        case StakingMode.Unstake:
+          setActionLabel(t('title.unstake'))
+          _header.subtitle = t('title.stakingModeNfts.unstakeHeaderSubtitle')
+          setHeader(_header)
+          setNfts(
+            loadingWalletStakedNfts ??
+              ({ loading: false, errored: true } as LoadingDataWithError<
+                NftCardInfo[]
+              >)
+          )
+          break
+        default:
+          break
+      }
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    []
```

Comment:
this will lead to this callback function only being able to reference the initial versions of variables. it sometimes makes sense to ignore the dependency array for `useEffect`s like you did below, but I think it's rare you'd want to do this here.